KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/RegexQuery.java

```diff
@@ -273,44 +273,44 @@ public QueryResults performQuery() throws NoOpenCoreException {
                         hit = hit.replaceAll("[^0-9]$", "");
                     }
 
-                    if (originalKeyword.getArtifactAttributeType() == BlackboardAttribute.ATTRIBUTE_TYPE.TSK_EMAIL) {
-                        // Reduce false positives by eliminating email address hits that are either
-                        // too short or are not for valid top level domains.
-                        if (hit.length() < MIN_EMAIL_ADDR_LENGTH
-                                || !DomainValidator.getInstance(true).isValidTld(hit.substring(hit.lastIndexOf('.')))) {
-                            continue;
-                        }
-                    }
+                    switch (originalKeyword.getArtifactAttributeType()) {
+                        case TSK_EMAIL:
+                            /*
+                             * Reduce false positives by eliminating email
+                             * address hits that are either too short or are not
+                             * for valid top level domains.
+                             */
+                            if (hit.length() >= MIN_EMAIL_ADDR_LENGTH
+                                    && DomainValidator.getInstance(true).isValidTld(hit.substring(hit.lastIndexOf('.')))) {
+                                addHit(content, snippet, hitMatcher, hit, hits, docId);
+                            }
```

Comment:
@millmanorama , in conversation you wondered whether the switch is better than the if-else blocks. FWIW, I like the switch. I think it is an improvement to code clarity.